test/benchmarks/include/vikunja/bench/memory.hpp

```diff
@@ -0,0 +1,213 @@
+                TIdx const threadLastElemIdx(threadFirstElemIdx + threadElemExtent);
+                TIdx const threadLastElemIdxClipped(alpaka::math::min(acc, numElements, threadLastElemIdx));
+
+                for(TIdx i(threadFirstElemIdx); i < threadLastElemIdxClipped; ++i)
+                {
+                    output[i] = m_begin + static_cast<TData>(i) * m_increment;
+                }
+            }
+        }
+    };
+
+
+    //! Allocates memory and initialises each value with `begin + index * increment`,
+    //! where index is the position in the output vector. The allocation is done with `setup.devAcc`.
+    //!
+    //! \tparam TData Data type of the memory buffer.
+    //! \tparam TSetup Fully specialized type of `vikunja::test::TestAlpakaSetup`.
+    //! \tparam Type of the extent.
+    //! \tparam TBuf Type of the alpaka memory buffer.
+    //! \param setup Instance of `vikunja::test::TestAlpakaSetup`. The `setup.devAcc` and `setup.queueDev` is used
+    //! for allocation and initialization of the the memory.
+    //! \param extent Size of the memory buffer. Needs to be 1 dimensional.
+    //! \param begin Value of the first element. Depending of TData, it can be negative.
+    //! \param increment Distance between two elements of the vector. If the value is negative, the value of an
+    //! element is greater than its previous element.
+    template<
+        typename TData,
+        typename TSetup,
+        typename TExtent,
+        typename TBuf = alpaka::Buf<typename TSetup::DevAcc, TData, alpaka::DimInt<1u>, typename TSetup::Idx>>
+    TBuf allocate_mem_iota(
+        TSetup& setup,
+        TExtent const& extent,
+        TData const begin = TData{0},
+        TData const increment = TData{1})
+    {
+        // TODO: test also 2 and 3 dimensional memory
+        static_assert(TExtent::Dim::value == 1);
+
+        // TODO: optimize utilization for CPU backends
+        typename TSetup::Idx const elementsPerThread = 1;
+        typename TSetup::Idx linSize = extent.prod();
+
+        TBuf devMem(alpaka::allocBuf<TData, typename TSetup::Idx>(setup.devAcc, extent));
+
+        alpaka::WorkDivMembers<typename TSetup::Dim, typename TSetup::Idx> const workDiv(
+            alpaka::getValidWorkDiv<typename TSetup::Acc>(
+                setup.devAcc,
+                extent,
+                elementsPerThread,
+                false,
+                alpaka::GridBlockExtentSubDivRestrictions::Unrestricted));
+
+        IotaFunctor iotaFunctor(begin, increment);
+
+        alpaka::exec<typename TSetup::Acc>(
+            setup.queueAcc,
+            workDiv,
+            iotaFunctor,
+            alpaka::getPtrNative(devMem),
+            linSize);
+
+        return devMem;
+    }
+
+    template<typename TData>
+    class ConstantInitFunctor
+    {
+    private:
+        TData const m_constant;
+
+    public:
+        //! Functor to write constant value in each element of a vector.
```

Comment:
```suggestion
        //! Functor to write a constant value into each element of a vector.
```

Also, shouldn't this be part of the class documentation instead of the constructor documentation?